src/lib/managers/CommandManager.ts

```diff
@@ -13,6 +13,7 @@ export class CommandManager extends Collection<string, Command> {
 			if (this.client) command.initialize(this.client);
 			if (Plugins.currentlyLoading) command.owner = Plugins.currentlyLoading;
 			this.set(command.name, command);
+			this.client.emit(Events.COMMAND_REGISTERED, command);
```

Comment:
The availability of the client is not guaranteed here. This will throw an error in some cases. (Same issue for similar places). Using the client also completely removes the purpose of our own logger.

Reply:
After all, above you said to add these events directly to discord.js, and not to logger.on("something")

Reply:
So should I put the events back into the logger?

Reply:
> 

I did not say that. 

And yes, the events should be put back in the logger